app/views/static_pages/room.html.erb

```diff
@@ -57,25 +55,26 @@ var comm = new Icecomm('9gv3ODZorBS2kVuNwnugBeV432PcNX3Qn0HmW33ofxK/u5Xolu');
 
     comm.on('disconnect', function(options) {
       document.getElementById(options.callerID).remove();
-
+      comm.close();
     });
+
     comm.on('data', function(options) {
     console.log("data: ",options.data);
-    //appendNewIM(options);
+    appendNewIM(options);
   });
 
-   comm.on('data',appendNewIM);
-
-  // var user_name = prompt("Your user name?");
-
   var $sendButton = $("#js-send-button-im");
   var $inputIM    = $("#js-input-im");
   var $chatBody   = $("#js-chat-body");
-  var clientOptions;
 
   $sendButton.on('click', sendIM);
   $inputIM.on('keypress', checkEnterKey);
 
+  //With Turbolinks pages will change without a full reload, so you can't rely on the usual window.beforepageunload event. Instead Turbolinks fires events on document to provide hooks into the lifecycle of the page.
+  $(document).on("page:fetch", function(){comm.close()}); //starting to fetch a new target page
+
```

Comment:
Lines 73-75 ensure that your local audio/vidio stream is stopped when you click on a link in rails using Turbolinks. Turbolinks fires events on document to provide hooks into the lifecycle of the page. 
page:fetch is an event indicating that we are starting to fetch a new target page from the server. documentation: https://github.com/rails/turbolinks